client/components/gravatar/index.jsx

```diff
@@ -94,4 +95,5 @@ export class Gravatar extends Component {
 
 export default connect( ( state, ownProps ) => ( {
 	tempImage: getUserTempGravatar( state, get( ownProps, 'user.ID', false ) ),
+	user: getCurrentUser( state ),
```

Comment:
Caveat, according to a quick `grep`, the `Gravatar` component is used in a couple of other places, not only `Layout`, where we're still passing a `user` prop explicitly. This change to `connect()` will override that passed prop, which can lead to developer confusion :wink: and subtle errors. On a conceptual level, we might also be using the `Gravatar` component to display the avatars of users other than the currently logged in one (maybe in a given site's user management -- the 'People' setion?)

I'd suggest removing the `connect()`ion to `user` here and move it to `Layout` instead -- essentially making it a drop-in replacement for what previously was using `lib/user` there, while continuing to pass on that `user` prop to child components. I think this would retain previous behavior as much as possible.

Reply:
Thanks. Good catch! 

I moved `getCurrentUser` into `MasterbarLoggedIn` instead of `Layout`.